tests/foreman/endtoend/test_cli_endtoend.py

```diff
@@ -0,0 +1,406 @@
+            user,
+            ActivationKey,
+            {
+                u'content-view-id': content_view['id'],
+                u'lifecycle-environment-id': lifecycle_environment['id'],
+                u'name': gen_alphanumeric(),
+                u'organization-id': org['id'],
+            }
+        )
+
+        # step 2.14: Add the products to the activation key
+        subscription_list = Subscription.with_user(
+            user['login'],
+            user['password']
+        ).list(
+            {u'organization-id': org['id']},
+            per_page=False
+        )
+        for subscription in subscription_list:
+            if subscription['name'] == DEFAULT_SUBSCRIPTION_NAME:
+                ActivationKey.with_user(
+                    user['login'],
+                    user['password']
+                ).add_subscription({
+                    u'id': activation_key['id'],
+                    u'quantity': 1,
+                    u'subscription-id': subscription['id'],
+                })
+
+        # step 2.14.1: Enable product content
+        ActivationKey.with_user(
+            user['login'],
+            user['password']
+        ).content_override({
+            u'content-label': 'rhel-6-server-rhev-agent-rpms',
+            u'id': activation_key['id'],
+            u'organization-id': org['id'],
+            u'value': '1',
+        })
+
+        # BONUS: Create a content host and associate it with promoted
+        # content view and last lifecycle where it exists
+        content_host = self._create(
+            user,
+            ContentHost,
+            {
+                u'content-view-id': content_view['id'],
+                u'lifecycle-environment-id': lifecycle_environment['id'],
+                u'name': gen_alphanumeric(),
+                u'organization-id': org['id'],
+            }
+        )
+        # check that content view matches what we passed
+        self.assertEqual(
+            content_host['content-view'],
+            content_view['name']
+        )
+        # check that lifecycle environment matches
+        self.assertEqual(
+            content_host['lifecycle-environment'],
+            lifecycle_environment['name']
+        )
+
+        # step 2.15: Create a new libvirt compute resource
+        self._create(
+            user,
+            ComputeResource,
+            {
+                u'name': gen_alphanumeric(),
+                u'provider': u'Libvirt',
+                u'url': u'qemu+tcp://{0}:16509/system'.format(
+                    settings.server.hostname),
+            }
+        )
+
+        # step 2.16: Create a new subnet
+        subnet = self._create(
+            user,
+            Subnet,
+            {
+                u'name': gen_alphanumeric(),
+                u'network': gen_ipaddr(ip3=True),
+                u'mask': u'255.255.255.0',
+            }
+        )
+
+        # step 2.17: Create a new domain
+        domain = self._create(
```

Comment:
I've followed basically the same code style for these calls just to maintain some pattern. But I can check which ones fill in one line and update them.